untyped-lambda-calculus/article.md

```diff
@@ -0,0 +1,489 @@
+# Untyped Lambda Calculus
+
+When talking about Haskell, the term "lambda calculus" often crops up.
+It is a theoretical framework used to define the meaning of computation in many functional languages.
+Understanding lambda calculus can be very helpful when talking about functional programs, e.g. those written in Haskell, Agda, Idris, etc.
+Perhaps surprisingly, understanding lambda calculus can be also helpful in understanding C++ template metalanguage, because it's a functional language in its own right (although not, strictly speaking, based on lambda calculus).
+Lambda calculus is also used extensively in programming language theory research and functional language design.
+
+There are two broad kinds of lambda calculus: _untyped_ and _typed_.
+The _untyped_ kind, discussed in this article, expresses an unrestricted computation.
```

Comment:
This needs one or two more sentences, probably one with information like this: 

"the untyped kind doesn't find much use beyond the computability theory.
Still, it's necessary to understand the untyped lambda calculus to meaningfully discuss the typed variety (or, more accurately, varieties)."

Reply:
Done.